### PR TITLE
Doc update and cleanup

### PR DESCRIPTION
**Pull request by MrTroble**
This removes a lot of unnecessary docs stuff from the data and adds auto generation for docs

### CODE REVIEW

File: TGEngine/io/Properties.hpp [hunk 1]
```diff
@@ -12,40 +12,35 @@ namespace prop {
...
 
 	/*
 	 * Property storage -> only for tiny bits of data
 	 */
-	SINCE(0, 0, 1)
-		class Properties {
+	class Properties {
 		public:
 			/*
 			 * Gets the value or returns 0. false or an empty string
```
Comment (IceDynamix) [inline TGEngine/io/Properties.hpp]:
```suggestion
			 * Gets the value or returns 0, false or an empty string
```

File: TGEngine/pipeline/Instance.hpp [hunk 2]
```diff
@@ -7,21 +7,17 @@ extern VkResult lastResult;
 extern VkInstance instance;
 extern VkDebugUtilsMessengerEXT debugMessager;
 
-SINCE(0, 0, 1)
 void createInstance();
 
+#ifdef DEBUG
 /*
  * This defines the debug callback for the messages of the vulkan debug extension
```
Comment (IceDynamix) [inline TGEngine/pipeline/Instance.hpp]:
```suggestion
 * This defines the debug callback for the messages of the Vulkan debug extension
```

File: TGEngine/pipeline/window/Window.hpp [hunk 3]
```diff
@@ -61,31 +60,25 @@ extern std::vector<Window*> windowList;
 /*
  * Creates a window based on the properties
  */
-SINCE(0, 0, 1)
 void createWindow(Window* window);
 
 /*
  * Creates the actual window
  * Attention! Much platform code here!
```
Comment (IceDynamix) [inline TGEngine/pipeline/window/Window.hpp]: What do you mean by that? Please elaborate

File: TGEngine/io/Files.hpp [hunk 4]
```diff
@@ -36,29 +36,25 @@ namespace tge {
...
 		/*
 		 * Opens the file
 		 * (In debug mode it checks if the open was successfull)
 		 */
-		SINCE(0, 0, 4)
-			File open(char* name, char* mode);
+		File open(char* name, char* mode);
 
 		/*
 		 * Opens the file and get's the size of the file
```
Comment (IceDynamix) [inline TGEngine/io/Files.hpp]:
```suggestion
		 * Opens the file and gets the size of the file
```

File: TGEngine/io/Files.hpp [hunk 5]
```diff
@@ -36,29 +36,25 @@ namespace tge {
...
 		 * (In debug mode it checks if the open was successfull)
```
Comment (IceDynamix) [inline TGEngine/io/Files.hpp]:
```suggestion
		 * (In debug mode it checks if the open was successful)
```

File: TGEngine/io/Files.hpp [hunk 6]
```diff
@@ -36,29 +36,25 @@ namespace tge {
...
-		 * (the long* needs to be vailid pointer, will be checked in debug mode)
+		 * (the long* needs to be valid pointer, will be checked in debug mode)
 		 */
-		SINCE(0, 0, 4)
-			File readFileSize(char* name, char* mode, /*OUT*/ long* fileLength);
+		File readFileSize(char* name, char* mode, /*OUT*/ long* fileLength);
 
 		/*
 		 * Reads all bytes out of a file
```
Comment (IceDynamix) [inline TGEngine/io/Files.hpp]:
```suggestion
		 * Reads all bytes of a file
```

File: TGEngine/io/Files.hpp [hunk 7]
```diff
@@ -36,29 +36,25 @@ namespace tge {
...
 		 * (In debug mode it checks if the open was successfull)
```
Comment (IceDynamix) [inline TGEngine/io/Files.hpp]:
```suggestion
		 * (In debug mode it checks if the open was successful)
```

File: TGEngine/io/Files.hpp [hunk 8]
```diff
@@ -36,29 +36,25 @@ namespace tge {
 		 * Use "current_working_dir" to get the cwd after quereing.
 		 * (Is automatically called in startTGEngine())
 		 */
-		SINCE(0, 0, 3)
-			void initFileSystem();
+		void initFileSystem();
 
 		/*
 		 * Opens the file
 		 * (In debug mode it checks if the open was successfull)
```
Comment (IceDynamix) [inline TGEngine/io/Files.hpp]:
```suggestion
		 * (In debug mode it checks if the open was successful)
```

File: TGEngine/io/Files.hpp [hunk 9]
```diff
@@ -36,29 +36,25 @@ namespace tge {
 		 * Use "current_working_dir" to get the cwd after quereing.
```
Comment (IceDynamix) [inline TGEngine/io/Files.hpp]: Do you mean queueing? querying?

File: TGEngine/util/Math.hpp [hunk 10]
```diff
@@ -8,26 +8,4 @@
...
-
-	SINCE(0, 0, 1)
-		std::uint32_t u_max(std::uint32_t val1, std::uint32_t val2);
-
-	SINCE(0, 0, 1)
-		std::uint32_t u_min(std::uint32_t val1, std::uint32_t val2);
-
-}
\ No newline at end of file
+#define TGE_MAX(a,b) (a > b ? a:b)
```
[the remark this answers is not in the document]
Comment (MrTroble) [reply TGEngine/util/Math.hpp]: The primarily  purpose of this is to make it platform independent. As the basic math functions might only be defined by the Microsoft standard library. As there is no standard for Macros

File: docs/docs/shadertool.html [hunk 11]
```diff
@@ -3,17 +3,18 @@
...
 <body>
 
     <h3>Documentations - Shader Tool</h3>
 
     <div class="menu">
-        <a href="./index.html">Home</a>
-        <a href="./docs.html">Docs</a>
+        <a href="../index.html">Home</a>
+        <a href="index.html">Docs</a>
+        <a href="gmc.html">Gamecontent</a>
```
[the remark this answers is not in the document]
Comment (MrTroble) [reply docs/docs/shadertool.html]: This shouldn't be in there at all

File: TGEngine/pipeline/Swapchain.cpp [hunk 12]
```diff
@@ -16,7 +16,7 @@ void createSwapchain() {
 	lastResult = vkGetPhysicalDeviceSurfaceCapabilitiesKHR(physicalDevice, win->surface, &surface_capabilities);
 	HANDEL(lastResult)
 
-		imagecount = math::u_min(math::u_max(imagecount, surface_capabilities.minImageCount), surface_capabilities.maxImageCount);
+		imagecount = TGE_MIN(TGE_MIN(imagecount, surface_capabilities.minImageCount), surface_capabilities.maxImageCount);
```
[the remark this answers is not in the document]
Comment (MrTroble) [reply TGEngine/pipeline/Swapchain.cpp]: This was clearly a bug thanks!

File: TGEngine/io/Files.hpp (hunk 9 above)
Comment (MrTroble) [reply TGEngine/io/Files.hpp]: querying ...

**Comment by MrTroble** [comment]
[Documentation](https://trello.com/c/zhpfwpQc/35-documentation)

**Pull request merged**